src/app/compose/compose.component.ts

```diff
@@ -109,18 +109,22 @@ export class ComposeComponent implements AfterViewInit, OnDestroy, OnInit {
         });
     }
 
-    ngOnInit() {
+    async ngOnInit() {
         if (this.model.isUnsaved()) {
             this.editing = true;
             this.isUnsaved = true;
             this.shouldReturnToPreviousPage = true;
-            const from: FromAddress = this.draftDeskservice.froms.find((f) =>
+            this.has_pasted_signature = false;
+            
+            const defaultProfile = await this.rmmapi.getDefaultProfile().toPromise();
+            const defaultProfileIndex = await this.draftDeskservice.froms.findIndex(profile => profile.email === defaultProfile.email);
+
+            const from: FromAddress = this.draftDeskservice.froms.find((f) => 
                 f.nameAndAddress === this.model.from || f.email === this.model.from);
 
-            this.has_pasted_signature = false;
-            if (!from) {
+            if (!from || this.router.url.includes('compose?new=true')) {
```

Comment:
When is this second check needed?

Reply:
Actually now it is. For some reason "this.model.from" isn't empty from the start so "!from" isn't enough, because there is always something inside "from". I believe it needs some more digging to see where in the app the model gets values assigned, but it's a temporary fix until I figure out what's going on there. 

Probably should've said all of that right from the start :D

Reply:
I think you should be fixing the proper "from" here: https://github.com/runbox/runbox7/blob/master/src/app/compose/draftdesk.service.ts#L58